src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const char of actions) {
+    if (char['type'] === 'addProperties') {
+      for (const keys in char['extraData']) {
+        state[keys] = char['extraData'][keys];
+      }
+    }
+
+    if (char['type'] === 'removeProperties') {
+      for (const keys of char['keysToRemove']) {
+        delete state[keys];
+      }
```

Comment:
u get one key from all keys
```suggestion
      for (const key of char['keysToRemove']) {
        delete state[key];
      }
```